pkg/util/backuputil/backuptutil.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backuputil
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"path"
+
+	api "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+	"github.com/coreos/etcd-operator/pkg/util/constants"
+	"github.com/coreos/etcd-operator/pkg/util/etcdutil"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/coreos/etcd/clientv3"
+	"k8s.io/api/core/v1"
+)
+
+func NewAWSConfig(secret *v1.Secret, tmpDir string) (*session.Options, error) {
+	dir, err := ioutil.TempDir(tmpDir, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create dir %v: (%v)", dir, err)
+	}
+
+	options := &session.Options{}
+	options.SharedConfigState = session.SharedConfigEnable
+
+	creds := secret.Data[api.AWSSecretCredentialsFileName]
+	if len(creds) != 0 {
+		credsFile := path.Join(dir, "credentials")
+		err := ioutil.WriteFile(credsFile, creds, 0600)
+		if err != nil {
+			return nil, fmt.Errorf("failed to write credentials file: (%v)", err)
+		}
+		options.SharedConfigFiles = append(options.SharedConfigFiles, credsFile)
+	}
+
+	config := secret.Data[api.AWSSecretConfigFileName]
+	if config != nil {
+		configFile := path.Join(dir, "config")
+		err := ioutil.WriteFile(configFile, config, 0600)
+		if err != nil {
+			return nil, fmt.Errorf("failed to write config file: (%v)", err)
+		}
+		options.SharedConfigFiles = append(options.SharedConfigFiles, configFile)
+	}
+	return options, nil
+}
+
+func GetMemberWithMaxRev(pods []*v1.Pod, tc *tls.Config) (*etcdutil.Member, int64) {
```

Comment:
document public funcs. and this func probably should not live inside backuputil. it is not backup related.

Reply:
let me think about a better place to place it